File: app/src/main/java/yangFenTuoZi/runner/plus/service/callback/InstallTermExtCallback.java
package yangFenTuoZi.runner.plus.service.callback;

public class InstallTermExtCallback {
    private final IInstallTermExtCallback mCallback;

    public InstallTermExtCallback(IInstallTermExtCallback callback) {
        mCallback = callback;
    }

    public void onMessage(String message) {
        try {
            mCallback.onMessage(message);
        } catch (Throwable ignored) {
        }
    }

    public void onExit(boolean isSuccessful) {
        try {
            mCallback.onExit(isSuccessful);
        } catch (Throwable ignored) {
        }
    }
}

File: app/src/main/java/yangfentuozi/runner/service/callback/InstallTermExtCallback.kt
package yangfentuozi.runner.service.callback

import android.os.RemoteException

class InstallTermExtCallback(private val mCallback: IInstallTermExtCallback?) {
    fun onMessage(message: String?) {
        try {
            mCallback?.onMessage(message)
        } catch (_: RemoteException) {
        }
    }

    fun onExit(isSuccessful: Boolean) {
        try {
            mCallback?.onExit(isSuccessful)
        } catch (_: RemoteException) {
        }
    }
}